dedupe/variables/ilcs.py

```diff
@@ -0,0 +1,55 @@
+import string
+import math
+import functools
+
+import ilcs_parser
+from parseratorvariable import ParseratorType, consolidate
+
+CITATION = (
+    ('chapter', ('Chapter',)),
+    ('act prefix', ('ActPrefix',)),
+    ('section', ('Section',)),
+    ('subsection', ('SubSection',)),
+    ('attempted', ('Attempted',))
+)
+
+
+class ILCSType(ParseratorType):
+    type = 'ILCS'
+
+    def tagger(self, field):
+        return self.tag(field)
+
+    def __init__(self, definition):
+        self.components = (('Citation', self.compareFields, CITATION),)
+        block_parts = ('Citation',)
+        super().__init__(definition, ilcs_parser.tag, block_parts)
+
+    def compareFields(self, parts, field_1, field_2):
+        """
+        Override the base comparison function to provide more semantically
+        meaningful comparisons of field_1 and field_2, depending on the type
+        of the field.
+        """
+        joinParts = functools.partial(consolidate, components=parts)
+        for part, (part_1, part_2) in zip(parts, zip(*map(joinParts, [field_1, field_2]))):
+            if part == ('Attempted',):
+                # We expect Attempted indicators like "(att)" to be strings
+                yield self.compareString(part_1, part_2)
+            else:
+                if part_1.isdigit() and part_2.isdigit():
```

Comment:
i'd tend to compare things either as strings (with there idea that there might be typos) or as exact matches. 

i think that citations that are close by are not ones we want tot match.

Reply:
Yeah, I think that makes sense. I'll give it a try with string comparisons and see if we get better performance. Do you think there's a way to merge the exact match/string approaches, or should we try one or the other?

Reply:
For the string comparison it seems like we actually won't need to override `compareFields`, since part-by-party comparison is the default behavior of `ParseratorType`:

https://github.com/dedupeio/parseratorvariable/blob/bb8808b05418300bcf42d88bc4e8570a65011ffa/parseratorvariable/__init__.py#L161-L164

Reply:
> Yeah, I think that makes sense. I'll give it a try with string comparisons and see if we get better performance. Do you think there's a way to merge the exact match/string approaches, or should we try one or the other?

You can just compare it both ways and create two entries in the vector for each.

Reply:
Gotcha. String comparison didn't improve the accuracy by much so I'm going to try adding on exact match to the vector too. To do that I'm thinking of updating `__init__` to add 1 to `expanded_size` ([source](https://github.com/dedupeio/parseratorvariable/blob/0d5100eeaf42b889c865bb5cf09b92ccfaa3b660/parseratorvariable/__init__.py#L43-L46)) and append a field definition for the exact match by [overriding `fields()`](https://github.com/dedupeio/parseratorvariable/blob/0d5100eeaf42b889c865bb5cf09b92ccfaa3b660/parseratorvariable/__init__.py#L142-L159). Is that it, or can you think of anything else I'd need to do? E.g. do I need to update [`compare()`](https://github.com/dedupeio/parseratorvariable/blob/0d5100eeaf42b889c865bb5cf09b92ccfaa3b660/parseratorvariable/__init__.py#L93)?

Reply:
that sounds right.